lib/spot-checker.js

```diff
@@ -160,126 +142,82 @@ export default class Spark {
     stats.endAt = new Date()
   }
 
-  async submitMeasurement (task, stats) {
-    console.log('Submitting measurement...')
-    const payload = {
-      sparkVersion: SPARK_VERSION,
-      zinniaVersion: Zinnia.versions.zinnia,
-      ...task,
-      ...stats,
-      participantAddress: Zinnia.walletAddress,
-      stationId: Zinnia.stationId
-    }
-    console.log('%o', payload)
-    const res = await this.#fetch('https://api.filspark.com/measurements', {
-      method: 'POST',
-      body: JSON.stringify(payload),
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      signal: AbortSignal.timeout(10_000)
-    })
-    await assertOkResponse(res, 'Failed to submit measurement')
-    const { id } = await res.json()
-    console.log('Measurement submitted (id: %s)', id)
-    return id
-  }
-
-  async nextRetrieval () {
-    const retrieval = await this.getRetrieval()
-    if (!retrieval) {
-      console.log('Completed all tasks for the current round. Waiting for the next round to start.')
-      return
-    }
-
+  async spotCheck (retrieval, dagScope, entityBytesRange) {
     const stats = newStats()
+    await this.executeSpotCheck({ retrieval, stats, dagScope, entityBytesRange })
 
-    await this.executeRetrievalCheck(retrieval, stats)
-
-    const measurementId = await this.submitMeasurement(retrieval, { ...stats })
     Zinnia.jobCompleted()
-    return measurementId
+    return stats
   }
 
-  async run () {
-    while (true) {
-      const started = Date.now()
+  async run ({ roundNum, maxTasks, minerId, dagScope = 'entity', entityBytesRange = '0:200' }) {
+    const results = []
+    const retrievalTasks = await this.#tasker.getRetrievalTasks({ roundNum, maxTasks, minerId, entityBytesRange })
+    for (const retrieval of retrievalTasks) {
```

Comment:
A "retrieval" and a "retrieval task" are different things. One is when the spot checker downloads something, the other is a task definition.

What about

```
for (const task of retrievalTask)
```